src/dmd/root/timeit.d

```diff
@@ -0,0 +1,79 @@
+/**
+ * Development utility for tracking the amount of time a section of code takes.
+ *
+ * Copyright:   Copyright (C) 1999-2021 by The D Language Foundation, All Rights Reserved
+ * Authors:     Stefan Koch
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/root/timeit.d, _timeit.d)
+ * Documentation:  https://dlang.org/phobos/dmd_root_timeit.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/root/timeit.d
+ */
+
+module dmd.root.timeit;
+
+/// Accumulates the ticks for this section of code into the provided counter
+struct TimeIt
+{
+    ulong *ticks_var;
+    ulong begin;
+
+    this(ulong* var)
+    {
+        pragma(inline, true);
+        ticks_var = var;
+        begin = tsc();
+    }
+
+    ~this()
+    {
+        pragma(inline, true);
+        *ticks_var += (tsc() - begin);
+    }
+}
+
+version (D_InlineAsm_X86_64)
+{
+    extern (C) ulong tsc()
```

Comment:
Why is this extern C?

Reply:
So I can be sure of the ABI.

Reply:
though I guess D abi also puts the return in RAX

Reply:
This is quite blatant global namespace abuse and might easily become a problem for DMD as-a-library, if some other library decides to take the `tsc` symbol itself. `extern(D)` would work just fine.